### PR TITLE
Help written from file

### PR DESCRIPTION
**Pull request by DoruM**
This allows for additional help messages to be written from a csv file. This is in conjuction with the paramter "--broker" which is in Enterprise gateway. This parameter is parsed and executed in the launching script of the gateway, and not in the gateway itself

### CODE REVIEW

File: distribution/src/main/assembly/generic-bin.xml [hunk 1]
```diff
@@ -103,6 +103,7 @@
         <include>commons-cli:commons-cli</include>
         <include>org.json:json</include>
         <include>org.apache.maven.shared:maven-shared-utils</include>
+        <include>net.sf.opencsv:opencsv</include>
```
Comment (dpwspoon) [inline distribution/src/main/assembly/generic-bin.xml]: We need to add this to the notice.txt of the gateway and ee gateway now that we are using this

File: server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java [hunk 2]
```diff
@@ -107,19 +121,70 @@ public void run() {
...
+
+                while ((nextLine = script_options.readNext()) != null) {
+                    Option o = new Option(null, nextLine[0], Boolean.parseBoolean(nextLine[1]), nextLine[2]);
+                    all_opt.addOption(o);
+                }
+                script_options.close();
+            }
+
+        } catch (IOException e) {
+            System.out.println("hex");
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java]: Why the system.out with "hex"

File: server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java [hunk 3]
```diff
@@ -107,19 +121,70 @@ public void run() {
...
+            
+            if (!o1.getLongOpt().equals(SCRIPT_ARG)) {
+                all_opt.addOption(o1);
+            } 
+        }
+
+        // also, get the file that has documentation for the values parsed in the launching script
+        File helpScript = getScriptedOptsFile(cmd);
+        appendScriptedOptions(helpScript, all_opt);
+        helpFormatter.printHelp("gateway.start", all_opt, true);
```
Comment (dpwspoon) [inline server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java]: I don't think we should tie ourselves down to gateway.start

**Comment by dpwspoon** [comment]
Please add a test, it will make it easier to review.  This looks ok to me.  Consider replacing the `SCRIPT_ARG` with an env variable that populates it instead.  It might be a little cleaner usage of it...

File: server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java (hunk 2 above)
Comment (DoruM) [inline server/src/main/java/org/kaazing/gateway/server/GatewayCommandLineProcessor.java]: Removed.

File: server/pom.xml [hunk 4]
```diff
@@ -250,6 +250,11 @@
             <classifier>keystore</classifier>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>net.sf.opencsv</groupId>
+            <artifactId>opencsv</artifactId>
+            <version>2.3</version>
```
Comment (DoruM) [inline server/pom.xml]: Done.

File: distribution/src/main/assembly/generic-bin.xml (hunk 1 above)
Comment (DoruM) [inline distribution/src/main/assembly/generic-bin.xml]: Done. Added to gateway/NOTICE.txt, gateway.server/NOTICE.txt, enterprise.gateway.NOTICE.txt

**Comment by DoruM** [comment]
Changes added. Also unit test was added.

**Pull request merged**